src/Util/ModuleUtils.php

```diff
@@ -74,18 +103,53 @@ public function downloadMainClass(string $moduleName, Version $version): void
 
         $response = $this->client->get(sprintf(self::GITHUB_MAIN_CLASS_ENDPOINT, $moduleName, $version->getTag(), $moduleName));
         file_put_contents($path . '/' . $moduleName . '.php', $response->getBody());
-        $this->saveDownloadUrl($moduleName, $version);
     }
 
-    /**
-     * @return Module[]
-     */
-    public function getLocalModules(): array
+    public function download(string $moduleName, Version $version): void
+    {
+        $path = join('/', [$this->moduleDir, $moduleName, $version->getTag()]);
+        if (!is_dir($path)) {
+            mkdir($path, 0777, true);
+        }
+
+        if ($version->getGithubUrl() === null) {
+            throw new RuntimeException(sprintf('Unable to download %s %s because it has no Github url', $moduleName, $version->getTag()));
+        }
+
+        $response = $this->client->get($version->getGithubUrl());
+        file_put_contents($path . '/' . $moduleName . '.zip', $response->getBody());
```

Comment:
You need to check the path is valid before I think

Reply:
The path is created before if it doesn't exists